File: 392-is-subsequence/is-subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int l = 0;
        for (auto i = 0; i < t.size() && l < s.size(); i++) {
            if (t[i] == s[l])
                l++;
        }
        return l == s.size() ? true : false;
    }
};

File: 392-is-subsequence/is-subsequence.go
func isSubsequence(s string, t string) bool {
    first, second := 0, 0

    for first < len(s) && second < len(t) {
        if s[first] == t[second] {
            first++
        }
        second++
    }
    return first == len(s)
}